include/klee/CommandLine.h

```diff
@@ -44,7 +44,7 @@ enum QueryLoggingSolverType
  */
 extern llvm::cl::list<QueryLoggingSolverType> queryLoggingOptions;
 
-enum CoreSolverType { STP_SOLVER, METASMT_SOLVER, DUMMY_SOLVER };
+enum CoreSolverType { STP_SOLVER, METASMT_SOLVER, DUMMY_SOLVER, Z3_SOLVER };
```

Comment:
@MartinNowack 
Again I did it this way because this is how the other solvers are handled. Currently the style is the enums (and command line options) for all solver exists in all build configurations but if you try to use a solver that KLEE wasn't compiled with ``createCoreSolver()`` will emit message saying it wasn't compiled with support for that solver and cause KLEE or kleaver to abort.

The alternative style is those enums and command line options are added only if KLEE is built with support for that solver.

I think both methods are valid and I happy with either. @ccadar what do you think? If we do make this change I'd like it to be a separate PR that will be merged after this one because this PR is about adding Z3 support and not changing how all the solvers are handled.